beetmoverscript/utils.py

```diff
@@ -175,17 +175,21 @@ def generate_beetmover_template_args(context):
 def _generate_beetmover_template_args_maven(task, release_props):
     tmpl_args = {
         'artifact_id': task['payload']['artifact_id'],
-        'branch': release_props['branch'],
-        'buildid': release_props['buildid'],
+        'branch': release_props.get('branch'),
+        'buildid': release_props.get('buildid'),
         'product': release_props['appName'],
         'template_key': 'maven_{}'.format(release_props['appName']),
     }
 
-    # Change version number to major.minor.buildId because that's what the build task produces
-    version = FirefoxVersion.parse(task['payload']['version'])
-    tmpl_args['version'] = '{}.{}.{}'.format(
-        version.major_number, version.minor_number, tmpl_args['buildid']
-    )
+    # FIXME: find a better solution here
```

Comment:
@JohanLorenzo Not sure how to fix this to be honest. Is there any more relaxed `FirefoxVersion` that allows the kind of version that `android-components` needs? I'd really like some form of checks for the version. Right now there's none of it and that's not secure.

Reply:
The simplest solution is probably to ask android-components to use version numbers that match `MAJOR.MINOR.PATCH`. This way, `FirefoxVersion` works. 

Does this seem feasible to you @pocmo @csadilek?

Reply:
Oh, that'd be awesome if possible! 👍

Reply:
👍